src/lib.rs

```diff
@@ -1087,6 +1095,62 @@ impl EGraph {
         }
     }
 
+    fn calc_helper(
+        &mut self,
+        idents: Vec<IdentSort>,
+        exprs: Vec<Expr>,
+        depth: &mut i64,
+    ) -> Result<(), Error> {
+        let mut iter = exprs.into_iter();
+        if let Some(a) = iter.next() {
```

Comment:
Something goofy is happening with this loop structure. Could you refactor into a loop over [`exprs.windows (2)`](https://doc.rust-lang.org/stable/std/primitive.slice.html#method.windows)? Also no need to `if let` this whole thing, I think just bailing early if `exprs.len() < 2` would suffice.